### DISCUSSION TITLE
how to implement `/@username` route?

### DISCUSSION THREAD

**Discussion by limboy** [remix-run/remix#4035] [Q&A] [answered]
defined routes in `remix.config.js` like this, then visit `/@alice` it shows `404`, but if I replace `@` to `~`, it works.

``` js 
routes(defineRoutes) {
    return defineRoutes((route) => {
      route('/@:slug', 'routes/u/$slug.tsx');
    });
  },
```

**Comment by krall12** [comment]
Use just the `:slug` as the dynamic parameter and then verify the username starts with an @ symbol in the routes loader. 

```tsx
export const loader = async ({params}: LoaderArgs) => {
  // make sure it starts with an @
  if (!params.slug.startsWith('@')) {
    throw new Response('Not Found', { status: 404 }); 
  }

  // ...continue with the rest of the loader
}
```

**Reply by limboy** [reply]
does it work in your environment? I tried locally it doesn't work.

```js
routes(defineRoutes) {
    return defineRoutes((route) => {
      route('/:slug', 'routes/u/$slug.tsx');
    });
  },
```

in `routes/u/$slug.tsx`,  the `loader` function won't be triggered with url like `http://localhost:8888/@user`, but `/~user` works.

**Comment by machour** [comment, marked as answer]
@limboy we have an open issue for that https://github.com/remix-run/remix/issues/846

It have been fixed in React Router 6.4.0, but you will have to wait for the Remix update to leverage this new version.
There's a workaround described in the issue until then, you may want to try it.
